include/moveitem.h

```diff
@@ -189,57 +189,40 @@ inline int CountItem(const char* pcItemName, int usMin, int usInvSlots)
 /*********************************************************************************/
 // PackFind will fill a CItemLocation passed to it, or NULL if item not found
 //
-
-// finds an open slot in a bag that has items (not empty bags)
+// finds an open slot in a bag or inventory
 CItemLocation* PackFind(CItemLocation* pFreeSlot, ItemClient* pUnequipSlot)
 {
 	unsigned char  ucCurSize = 10;
-	unsigned short usSlot    = 0;
-
-	unsigned short usDecrement = 0;
+	ItemDefinition* pUnequipInfo = pUnequipSlot->GetItemDefinition();
 
-	// no user-specific slot given, find the first free bag slot
-	for (usSlot = InvSlot_FirstBagSlot; usSlot <= GetHighestAvailableBagSlot(); usSlot++)
+	for (unsigned short usSlot = InvSlot_FirstBagSlot; usSlot <= GetHighestAvailableBagSlot(); usSlot++)
 	{
 		if (ItemClient* pInvSlot = GetPcProfile()->GetInventorySlot(usSlot))
 		{
-			ItemDefinition* pItemInfo    = GetItemFromContents(pInvSlot);
-			ItemDefinition* pUnequipInfo = GetItemFromContents(pUnequipSlot);
+			ItemDefinition* pItemInfo = pInvSlot->GetItemDefinition();
```

Comment:
looking at this logic here: we would fail to return open base inventory slots, no? I know that's how the old algorithm does it as well, but that doesn't seem correct

Reply:
No this should be caught by the code in the previous comment.